xclim/indices/run_length.py

```diff
@@ -293,6 +294,160 @@ def last_run(
     return out
 
 
+def run_length_with_date(
+    da: xr.DataArray, window: int, date: str = "07-01", dim: str = "time",
+):
+    """Return the index of the last item of the last run of at least a given length.
+
+    Parameters
+    ----------
+    da : xr.DataArray
+      Input N-dimensional DataArray (boolean)
+    window : int
+      Minimum duration of consecutive run to accumulate values.
+    date:
+      The date that a run must include to be considered valid.
+    dim : str
+      Dimension along which to calculate consecutive run (default: 'time').
+
+    Returns
+    -------
+    out : xr.DataArray
+      Index (or coordinate if `coord` is not False) of last item in last valid run. Returns np.nan if there are no valid run.
+
+    """
+    include_date = datetime.strptime(date, "%m-%d").timetuple().tm_yday
+
+    def _crld(rlda, rlwindow, rldoy, rldim):
+        mid_index = np.where(rlda.time.dt.dayofyear == rldoy)[0]
+        if (
+            mid_index.size == 0
+        ):  # The date is not within the group. Happens at boundaries.
+            all_nans = xr.full_like(rlda.isel(time=0), np.nan)
+            all_nans.attrs = {}
+            return all_nans
+        end = first_run(
+            (~rlda).where(rlda.time >= rlda.time[mid_index][0]),
+            window=rlwindow,
+            dim=rldim,
+        )
+        beg = first_run(rlda, window=rlwindow, dim=rldim)
+        sl = end - beg
+        sl = xr.where(
+            beg.isnull() & end.notnull(), 0, sl
+        )  # If series is never triggered
+        sl = xr.where(
+            beg.notnull() & end.isnull(), rlda.time.size - beg, sl
+        )  # If series is not ended by end of resample time frequency
+        return sl.where(sl >= 0)
+
+    return _crld(rlda=da, rlwindow=window, rldoy=include_date, rldim=dim)
+
+
+def run_end_after_date(
+    da: xr.DataArray,
+    window: int,
+    date: str = "07-01",
+    dim: str = "time",
+    coord: str = "dayofyear",
+):
+    """Return the index of the last item of the last run after a given date.
```

Comment:
I believe this is more : The index of the first item after the end of a run after a given date" ?
Cause its not the "end of the last run" more than its the (day after the) end of the run that started before the given date.

Reply:
Fixed.